gamecontroller_db.cpp

```diff
@@ -10,16 +11,16 @@
 
 
 
+//Note: sdl gamecontrollerdb maps  a,b,x,y differently, so we need to swap each pair (a<->b, x<->y)
 static const char *sdlname_to_mister_idx[] = {
-
 	"dpright",
 	"dpleft",
 	"dpdown",
 	"dpup",
-	"a",
 	"b",
-	"x",
+	"a",
 	"y",
+	"x",
```

Comment:
won't this change have the buttons swapped since you're using the array to map?
`SYS_BTN_A` is `[4]`, `SYS_BTN_B` is `[5]`, `SYS_BTN_X` is `[6]` and `SYS_BTN_Y` is `[7]`

Reply:
Yes, and that is intended. SDL does a/b/x/y in the xbox layout style (a is south, b is east) . Mister's default/"main" uses the SNES/nintendo virtual gamepad style where a is east, b is south.